modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2530,15 +2546,13 @@ def sort_function(df):  # pragma: no cover
             )
 
         result = self._apply_func_to_range_partitioning(
-            key_columns=[columns[0]], func=sort_function, ascending=ascending, **kwargs
+            key_columns=[columns[0]],
+            func=sort_function,
+            ascending=ascending,
+            preserve_columns=True,
+            **kwargs,
         )
-
-        result.set_axis_cache(self.copy_axis_cache(axis.value ^ 1), axis=axis.value ^ 1)
         result.set_dtypes_cache(self.copy_dtypes_cache())
```

Comment:
Why didn't you move this cache setting?

Reply:
I thought that adding the `preserve_dtypes` parameter would be too much. Moreover, dtypes are independent of the implementation that was chosen underneath.

Reply:
ok